### PR TITLE
move assertions and iterators out of system.nim

### PR DESCRIPTION
**Pull request by narimiran**
First steps towards our v0.20 goal https://github.com/nim-lang/Nim/issues/10385

### CODE REVIEW

File: lib/system/assertions.nim [hunk 1]
```diff
@@ -0,0 +1,86 @@
+include "system/assertions_helpers"
```
Comment (Araq) [inline lib/system/assertions.nim]: This makes me nervous. Is this included only once, or does this lead to code duplication?

**Comment by dom96** [comment]
Do we have a plan anywhere of how this split is occurring? I don't want us to end up with a massively fragmented ``system``

File: lib/system/assertions.nim (hunk 1 above)
Comment (narimiran) [reply lib/system/assertions.nim]: It did lead to code duplication, so I redid that stuff.

File `system/assertions_helpers` (which had the duplicated code) is now gone.
Now there's a single file `system/fatal.nim` which is included [here](https://github.com/nim-lang/Nim/pull/10597/files#diff-e02ab787ce7af9352584fe91e2bf2b6dR3) and [here](https://github.com/nim-lang/Nim/pull/10597/files#diff-5cfdfeab79c2d2b7bbc78af7060c38c3R2884).

IMO, the PR is new much cleaner than in the original version.

File: lib/system/assertions.nim [hunk 2]
```diff
@@ -0,0 +1,89 @@
+include "system/helpers"
+
+when not declared(sysFatal):
```
Comment (Araq) [inline lib/system/assertions.nim]: Why is this ``when`` required?

Comment (narimiran) [reply lib/system/assertions.nim]: That's the way it is [currently done in system.nim](https://github.com/nim-lang/Nim/blob/devel/lib/system.nim#L3039) so I kept it the same.

**Comment by Araq** [comment]
> Do we have a plan anywhere of how this split is occurring? I don't want us to end up with a massively fragmented system

No, we tinker with it until it's acceptable. Please note that no system refactoring changes the interface.

**Pull request merged**